### ISSUE TITLE
Empty on demand listing after the first time

### ISSUE THREAD

**Issue by pandvan** [sterd71/plugin.video.eurosporton#9]
Using the latest version of the plugin (release 0.2.0) I've notice that after opening "On demand" section, if I come back to "On schedule" section, it's empty instead of showing the dates around the current day.
This happens even if I try to open "On demand" a second time. It works only the first one after a fresh Kodi startup.
The only way to make it works again is closing and opening Kodi again, just closing the plugin is not enough.
Enabling debug logs on Kodi doesn't help me to better understand what could be the cause.

**Comment by pandvan** [comment]
After some digging on this issue, it seems due to a Python 2.x bug on _datetime.strptime_ function.
Starting from the second execution of _onschedule_list_ function, if I print a stack trace, it shows on logs something like:
```
2020-11-28 14:05:13.981 T:140110255339072   ERROR:   File "/home/_my_user_/.kodi/addons/plugin.video.eurosporton/resources/lib/onschedule.py", line 54, in onschedule_list
2020-11-28 14:05:13.981 T:140110255339072   ERROR:     schedule_date = datetime.strptime(schedule_str, '%Y-%m-%d')
2020-11-28 14:05:13.981 T:140110255339072   ERROR: TypeError: attribute of type 'NoneType' is not callable
```
This happens even if _schedule_str_ is a proper unicode string, as in this case: _2020-11-21_

A similar issue is reported also in https://forum.kodi.tv/showthread.php?tid=112916 if it can help us to better understand what it's happening.

**Comment by Jordan23M** [comment]
> After some digging on this issue, it seems due to a Python 2.x bug on _datetime.strptime_ function.
> Starting from the second execution of _onschedule_list_ function, if I print a stack trace, it shows on logs something like:
> 
> ```
> 2020-11-28 14:05:13.981 T:140110255339072   ERROR:   File "/home/_my_user_/.kodi/addons/plugin.video.eurosporton/resources/lib/onschedule.py", line 54, in onschedule_list
> 2020-11-28 14:05:13.981 T:140110255339072   ERROR:     schedule_date = datetime.strptime(schedule_str, '%Y-%m-%d')
> 2020-11-28 14:05:13.981 T:140110255339072   ERROR: TypeError: attribute of type 'NoneType' is not callable
> ```
> 
> This happens even if _schedule_str_ is a proper unicode string, as in this case: _2020-11-21_
> 
> A similar issue is reported also in https://forum.kodi.tv/showthread.php?tid=112916 if it can help us to better understand what it's happening.

Hi, accidentally installing another addon, fixed the problem you describe above.

I am not an expert but the problem with Phyton is fixed with some update included in that addons version.
 
The addon i installed is the neflix addons  that include some update in phyton library.
CastagnaIT / plugin.video.netflix 

Hope it can be helpful, since then no more empty folder list.

Thanks for the great addons eurosport.

**Comment by robegrassi** [comment]
> A similar issue is reported also in https://forum.kodi.tv/showthread.php?tid=112916 if it can help us to better understand what it's happening.

@pandvan Thanks, I can confirm the workaround suggested in that thread does work. This is how I modified onschedule.py:

```
import time
...
try:
    schedule_date = datetime.strptime(schedule_str, '%Y-%m-%d') 
except TypeError:
    schedule_date = datetime.fromtimestamp(time.mktime(time.strptime(schedule_str, '%Y-%m-%d')))
```

**Comment by sterd71** [comment]
This fix has been implemented into release v0.2.1.
